src/openqaoa-core/openqaoa/problems/sherrington_kirkpatrick.py

```diff
@@ -0,0 +1,166 @@
+        mapping = dict(
+            zip(input_networkx_graph, range(input_networkx_graph.number_of_nodes()))
+        )
+        self._G = nx.relabel_nodes(input_networkx_graph, mapping)
+
+    @staticmethod
+    def random_instance(**kwargs):
+        """
+        Creates a random instance of the SK problem, whose graph is random following the Erdos-Renyi model.
+
+        Parameters
+        ----------
+        **kwargs:
+
+            n_nodes: int
+                The number of nodes (spins) in the graph.
+            mu: float
+                Mean value of the distribution for the J_{ij} values.
+            sigma: float
+                Standard deviation of the distribution for the J_{ij} values.
+            seed: int
+                Seed to be used for the random case.
+
+        Returns
+        -------
+            A random instance of the SK problem.
+        """
+        n_nodes = kwargs.get("n_nodes")
+        mu = kwargs.get("mu", 0)
+        sigma = kwargs.get("sigma", 1)
+        seed = kwargs.get("seed", None)
+        np.random.seed(seed)
+        
+        G = nx.Graph()
+        G.add_weighted_edges_from([[i, j, round(np.random.normal(mu, sigma),3)] for i in range(n_nodes) for j in range(i+1, n_nodes)])
+        
+        return SK(G)
+
+    @property
+    def docplex_model(self):
+        mdl = Model("SK_Problem")
+        num_spins = self.G.number_of_nodes()
+
+        x = {i: mdl.binary_var(name=f"x_{i}") for i in range(num_spins)}
+
+        energy = -1 * mdl.sum((2*x[i] - 1) * (2*x[j] - 1) * self.G[i][j]['weight']
+                         for i, j in self.G.edges())
+
+        mdl.minimize(energy)
+
+        return mdl
+
+    @property
+    def qubo(self):
+        """
+        Returns the QUBO encoding of this problem.
+
+        Returns
+        -------
+            The QUBO encoding of this problem.
+        """
+        cplex_model = self.docplex_model
+        qubo_docplex = FromDocplex2IsingModel(cplex_model).ising_model
+
+        return QUBO(
+            self.G.number_of_nodes(),
+            qubo_docplex.terms + [[]],
+            qubo_docplex.weights + [qubo_docplex.constant],
+            self.problem_instance,
+        )
+
+    def classical_solution(self, string: bool = False):
+        """
+        Solves the SK problem and returns the CPLEX solution.
+
+        Returns
+        -------
+        solution : dict
+            A dictionary containing the solution found with spin values.
+        """
+        model = self.docplex_model
+        model.solve()
+        status = model.solve_details.status
+        if status != 'integer optimal solution':
+            print(status) 
+        if string:
+            return "".join(str(round(model.solution.get_value(var))) for var in model.iter_binary_vars())
+        else:
+            return {var.name: model.solution.get_value(var) for var in model.iter_binary_vars()}
+
+    def plot_solution(self, solution, ax=None):
+        """
```

Comment:
I get the following error when doing this:
```
prob.plot_solution(q.result.lowest_cost_bitstrings()['solutions_bitstrings'][0])
```

![image](https://github.com/entropicalabs/openqaoa/assets/36878360/e7d1122d-b102-4856-8d6f-4ce27bf78198)

Reply:
fixed @shahidee44